src/language-js/printer-estree.js

```diff
@@ -3285,6 +3285,9 @@ function printFunctionParams(path, print, options, expandArg, printTypeParams) {
     !fun.rest;
 
   if (isFlowShorthandWithOneArg) {
+    if (options.arrowParens === "always") {
```

Comment:
I don't think that it's the right fix. This condition is there to --not-- print parenthesis. What happens if you put `options.arrowParens !== "always" &&` at the beginning of `isFlowShorthandWithOneArg` condition?

Reply:
```diff
  type f = (?arg) => void;
  
  class X {
    constructor(
-     ideConnectionFactory: (child_process$ChildProcess) => FlowIDEConnection = defaultIDEConnectionFactory
+     ideConnectionFactory: (
+       child_process$ChildProcess
+     ) => FlowIDEConnection = defaultIDEConnectionFactory
    ) {}
  }
  
  interface F {
-   ideConnectionFactoryLongLongLong: (child_process$ChildProcess) => FlowIDEConnection;
+   ideConnectionFactoryLongLongLong: (
+     child_process$ChildProcess
+   ) => FlowIDEConnection;
  }
```